### ISSUE TITLE
Have OSX support unified launch script

### ISSUE THREAD

**Issue by Tyriar** [Microsoft/vscode#2859]
The script added in https://github.com/Microsoft/vscode/pull/2650 needs its OSX support finished:

- [x] Determine where launcher.js should go on OSX
- [x] Ensure code.sh points to correct paths
- [x] Add action to install/uninstall global shortcut
- [x] Provide upgrade mechanism

Once this is done and `code.sh` is put into `/usr/bin` on installation, there will no longer be a need to add `function code () { VSCODE_CWD="$PWD" open -n -b "com.microsoft.VSCode" --args $*; }` to dotfiles as suggested by [the docs](https://code.visualstudio.com/Docs/editor/setup)

**Comment by joaomoreno** [comment]
:+1:

**Comment by joaomoreno** [comment]
Commit be4a1ef adds an executable `code` which allows me to do this:

```→ /Applications/Visual\ Studio\ Code\ Alpha.app/Contents/Resources/app/bin/code --help
Visual Studio Code v0.10.9-alpha

Usage: code [arguments] [paths...]

Options:
	-h, --help     Print usage.
	--locale       Use a specific locale.
	-n             Force a new instance of Code.
	-v, --version  Print version.
```

Pretty cool. Now we just need a way to place a symlink to this executable somewhere in the `PATH`.

**Comment by joaomoreno** [comment]
Fixed in d2a151f
